submission/models.py

```diff
@@ -1,56 +1,83 @@
+import hashlib
 import uuid
 
 from django.db import models
 
 
-# Central crosswalk model to connect all related records for a unique patient
 class Master(models.Model):
+    """
+    Central crosswalk model to connect all related records for a unique patient
+    """
 
     anon_patient_id = models.CharField(max_length=12, default=None, unique=True)
 
-# Submission record, ties all submission related info together.
-# A submission may be a single patient, but we may have multiple submissions for a single patient too
-# Once a hospital has verified the submission, we link back to the verified master record
-# identifier: some kind of user entered id
-# id_type:
-# insurance number, phone number, email, etc
+
 class Submission(models.Model):
+    """
+    Submission record, ties all submission related info together.
+    A submission may be a single patient, but we may have multiple submissions for a single patient too
+    Once a hospital has verified the submission, we link back to the verified master record
+    identifier: some kind of user entered id
+    id_type:
+    insurance number, phone number, email, etc
+    """
 
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    identifier = models.CharField(max_length=50, unique=True, null=False)
+    identifier = models.CharField(max_length=50, null=False)
     id_type = models.CharField(max_length=50, null=False)
-    master_id = models.ForeignKey(Master, on_delete=models.CASCADE, null=True)
+    master = models.ForeignKey(Master, on_delete=models.CASCADE, null=True, related_name='submissions')
+
+    @property
+    def patient_id(self):
+        return hashlib.md5(self.id_type.encode() + self.identifier.encode()).digest()[:12]
```

Comment:
Very cool but shouldn't we then enforce `id_type` to be a choice field? Makes the link more predictable I guess